Xkcd-Comics/ViewModels/ComicsViewModel.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  ComicsVM.swift
+//  Xkcd-Comics
+//
+//  Created by Jia-Jiuan Tsai on 2021-11-24.
+//
+
+import Foundation
+
+class ComicsViewModel: ObservableObject {
+    @Published private(set) var comic = Comic(num: 1, title: "Default", img: "Default", alt: "Default", year: "Default", month: "Default")
+    @Published private(set) var comics: [Comic] = []
```

Comment:
Duplicated state:

This app has:
1. Comics stored in UserDefaults
2. Comics stored in memory in this array

Then there is code that sort of tries to keep the two synced. When you favorite something, you add it to the memory array first, and then to UserDefaults. The code that does this is not entirely correct, so the two arrays will get out of sync. 

We need to have a *single source of truth*. You should never add something to one souce, then to another. Rather, when modifying, you should always modify the source of truth (UserDefaults in this case), and then fetch everything from there. That way, we are always in sync. 

In this specific case, we don't even need to have a memory `comics` array, because UserDefaults is local and fast to use. So we could just always check UserDefaults to see whether something is stored in favorites.